akka-distributed-data/src/main/scala/akka/cluster/ddata/GSet.scala

```diff
@@ -27,7 +28,8 @@ object GSet {
  * This class is immutable, i.e. "modifying" methods return a new instance.
  */
 @SerialVersionUID(1L)
-final case class GSet[A](elements: Set[A]) extends ReplicatedData with ReplicatedDataSerialization with FastMerge {
+final class GSet[A](val elements: Set[A], private[akka] val _delta: Option[GSet[A]] = None)
```

Comment:
OK, quite unfortunately https://issues.scala-lang.org/browse/SI-9547 disallows us from hiding generated curried apply. If I add single argument apply, this is ambiguous with the curried apply (see also https://issues.scala-lang.org/browse/SI-8735 ). If I don't, I would need to change the interface to always use curried form (`GSet()()`), which is undesired. Therefore the curried constructor needs to be made private and apply method stated in the Companion Object.